File: src/main/java/com/github/monosoul/git/updateindex/extended/ExtendedUpdateIndexCommand.java
package com.github.monosoul.git.updateindex.extended;

enum ExtendedUpdateIndexCommand {

    SKIP_WORKTREE("--skip-worktree"),
    NO_SKIP_WORKTREE("--no-skip-worktree"),
    MAKE_EXECUTABLE("--chmod=+x");

    private final String command;

    ExtendedUpdateIndexCommand(final String command) {
        this.command = command;
    }

    public String getCommand() {
        return command;
    }
}

File: src/main/kotlin/com/github/monosoul/git/updateindex/extended/ExtendedUpdateIndexCommand.kt
package com.github.monosoul.git.updateindex.extended

enum class ExtendedUpdateIndexCommand(val value: String) {
    SKIP_WORKTREE("--skip-worktree"),
    NO_SKIP_WORKTREE("--no-skip-worktree"),
    MAKE_EXECUTABLE("--chmod=+x"),
    MAKE_NOT_EXECUTABLE("--chmod=-x")
}